### PR TITLE
Cyclops adventurer class, for one-eyed humans and dwarfs

### PR DESCRIPTION
**Pull request by Hmeister-real**
Remaking this PR because the last one was bloated with 20 commits, and thats a little much. This one is almost the exact same as #97 except for some minor stat differences.

### CODE REVIEW

File: code/modules/jobs/job_types/roguetown/adventurer/types/combat/cyclops.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
+
+/datum/advclass/cyclops
+	name = "Cyclops"
+	tutorial = "What makes me a good Cyclops? If I were a bad Cyclops, I wouldn't be sittin' here, describin' it to you now would I? 'm a drunk, bloody cyclops."
```
Comment (AMyriad) [inline code/modules/jobs/job_types/roguetown/adventurer/types/combat/cyclops.dm:4]: lol

**Comment by JTGSZ** [comment]
I pinged you on github, the statline is if the monk and paladin fugged for a combo, and its at 6 slots. I'd like to see either the slots set to 2 so its actually special enough to justify the stats despite the flaw, or a slight lower of like 1 on swords and strength to justify the 6 slots.

**Comment by Hmeister-real** [comment]
> I pinged you too, the statline is if the monk and paladin fugged for a combo, and its at 6 slots. I'd like to see either the slots set to 2 so its actually special enough to justify the stats despite the flaw, or a slight lower of like 1 on swords and strength to justify the 6 slots. But this is just an opinion really.

I can tone down the slots, 6 could be a little much in retrospect. I want to keep the level 4 swords, but if it's too much i can definitely tone it down too.

**Comment by JTGSZ** [comment]
That is still 3 instead of 2 or 1 slots. The flaws system you created also requires someone to fill in new or remove flaws when it should generate from the character flaws list itself, which means someone will probably have to retouch this again later at some point.

File: code/__DEFINES/roguetown.dm [hunk 2]
```diff
@@ -5,6 +5,8 @@
 
 #define ALL_PATRON_NAMES_LIST		list("Astrata", "Noc", "Xylix", "Eora", "Malum", "Dendor", "Abyssor", "Necra", "Pestra", "Malum", "Ravox")
 
+#define ALL_FLAWS_LIST	list("alcoholic", "godfearing", "smoker", "junkie", "noeyer", "noeyel", "arm_r", "arm_l", "paranoid", "randflaw", "noflaw")
```
Comment (BlueMemesauce) [inline code/__DEFINES/roguetown.dm:8]:
```suggestion
```

File: code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm [hunk 3]
```diff
@@ -64,6 +64,8 @@ GLOBAL_LIST_EMPTY(billagerspawns)
 				if(A.amtchosen >= A.maxchosen)
 					testing("[A.name] fail9")
 					continue
+			if(!(H.charflaw in A.allowed_flaws))
```
Comment (JTGSZ) [inline code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm:67]: I took the time to actually review this. One thing to note, H.charflaw is actually a reference to a datum, you are attempting to check it against strings. The strings are also not the things within the name var in the datums.

File: code/__DEFINES/roguetown.dm (hunk 2 above)
Comment (Hmeister-real) [reply code/__DEFINES/roguetown.dm:8]: i wouldn't know how to, im not exactly a coder i just had someone on the discord walk me through the process

Comment (Hmeister-real) [reply code/__DEFINES/roguetown.dm:8]: well unless you mean theres already a list of character flaws and i just have to switch around the wording

File: code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm (hunk 3 above)
Comment (Hmeister-real) [reply code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm:67]: so would it not work? if it does i would rather have someone with more experience in coding try to optimize it in the future, becasue i have no clue how to fix that lol

Comment (JTGSZ) [reply code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm:67]: This wouldn't work in its current form, ya.

Comment (Hmeister-real) [reply code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm:67]: damn, well il draft this so it's not merged till someone can help fix then

File: code/__DEFINES/roguetown.dm (hunk 2 above)
Comment (BlueMemesauce) [reply code/__DEFINES/roguetown.dm:8]: yea theres already a global list of character flaws so we just need to remove this and change the var/list/allowed_flaws to use it

File: code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm [hunk 4]
```diff
@@ -11,6 +11,7 @@
 	"Dwarf",
 	"Dwarf"
 	)
+	var/list/allowed_flaws = ALL_FLAWS_LIST
```
Comment (BlueMemesauce) [inline code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm:14]:
```suggestion
	var/list/allowed_flaws = GLOB.character_flaws
```

File: code/modules/jobs/job_types/roguetown/adventurer/types/combat/cyclops.dm [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	allowed_sexes = list("male")
+	allowed_races = list("Humen",
+	"Dwarf",
+	)
+
+	allowed_flaws = list("noeyer", 
+	"noeyel"
+	)
```
Comment (BlueMemesauce) [inline code/modules/jobs/job_types/roguetown/adventurer/types/combat/cyclops.dm:12]:
```suggestion
	allowed_flaws = list("Cyclops (R)", 
	"Cyclops (L)"
	)
```

File: code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm (hunk 3 above)
Comment (BlueMemesauce) [inline code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm:67]:
```suggestion
			if(!(H.charflaw.name in A.allowed_flaws))
```

Comment (Hmeister-real) [reply code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm]: I think @BlueMemesauce just did it all, so re-opened.

**Comment by ChungusGamer666** [comment]
Fix build and I will merge this

**Comment by Hmeister-real** [comment]
ok the way it is right now has just 1 error, and i have actually no idea what is causing it. It could be a fluke error

**Comment by Hmeister-real** [comment]
> Fix build and I will merge this

it has no merge conflicts, and a single error thats probably insignificant

**Comment by BlueMemesauce** [comment]
You still have to fix the error, or the game won't compile

File: code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm [hunk 6]
```diff
@@ -11,6 +11,10 @@
 	"Dwarf",
 	"Dwarf"
 	)
+
+/datum/advclass/Initialize()
+     allowed_flaws = GLOB.character_flaws
```
Comment (BlueMemesauce) [inline code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm:16]: the error is that these are spaces instead of a tab

**Comment by Osmos123** [comment]
Question, are we going for an actual cyclops? https://ageofempires.fandom.com/wiki/Cyclops
image

I would be open for you to give them +10 strength and in return -8 intelligence with only 1-2 skill level in maces/swords/others. Makes it so easy to parry, but when a hit connects, they hit hard.

**Comment by NGGJamie** [comment]
The build on this PR  has been failing for several days now, is this still a project you're looking to pursue?

**Comment by BlueMemesauce** [comment]
he just needs to change spaces to a tab

**Comment by Hmeister-real** [comment]
been busy with other stuff, not abandoned this

**Comment by Hmeister-real** [comment]
> Question, are we going for an actual cyclops? https://ageofempires.fandom.com/wiki/Cyclops image
> 
> I would be open for you to give them +10 strength and in return -8 intelligence with only 1-2 skill level in maces/swords/others. Makes it so easy to parry, but when a hit connects, they hit hard.

no, just guys with one eye calling themselves cyclops + reference. I think actual cyclops would be a cool swamp/forest enemy, like launching people 20 tiles away and crushing their limbs

**Comment by Hmeister-real** [comment]
Single error is that Jammy isn't signed? The issue here is fixed, think its unrelated. Maybe its because the PR is too old or something, i can just remake it

**Pull request closed without merging**